### PR TITLE
fix(Datagrid): expandable row add background on row hover

### PR DESCRIPTION
**Pull request by matthewgallo**
Contributes to #3324 

Adds background color to expanded area when hovering on the expanded row.

#### What did you change?
```
packages/ibm-products/src/components/Datagrid/styles/_useExpandedRow.scss
```
#### How did you test and verify your work?
Storybook

### CODE REVIEW

**Review by elycheea — CHANGES_REQUESTED** [review]
If we’re following the row hover behavior from Carbon, I think hover over the expanded area should also show the hover state. Currently only seeing the hover state when hovering over the row itself but not the expanded content.

**Comment by matthewgallo** [comment]
Ah ok, I must have misread the issue. I'll change this ^

**Comment by matthewgallo** [comment]
Ended up taking a similar approach to Carbon for highlighting the parent row of the expanded area, couldn't get `:has(+ .expanded-class)` previous sibling selector to work here.

File: packages/ibm-products/src/components/Datagrid/Datagrid/DatagridExpandedRow.js [hunk 1]
```diff
@@ -15,8 +15,28 @@ const DatagridExpandedRow =
   (ExpandedRowContentComponent) => (datagridState) => {
     const { row } = datagridState;
     const { expandedContentHeight } = row || {};
+
+    const toggleParentHoverClass = (event, eventType) => {
+      if (
+        event &&
+        event.target &&
+        event.target.parentNode.previousElementSibling
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DatagridExpandedRow.js:23]: i'd do something like this to make this a little less painful
```
const parentNode = event?.target?.parentNode?.previousElementSibling;
if (parentNode)
...
```

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks good. just one small comment.

**Review by davidmenendez — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Review by elycheea — APPROVED** [review]

**Pull request merged**